### DISCUSSION TITLE
Problem with subscribe_data_change

### DISCUSSION THREAD

**Discussion by JuliaCordeiro** [FreeOpcUa/opcua-asyncio#1573] [Q&A]
We are using the `subscription.subscribe_data_change` method passing these two nodes:
- ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/TT36_10_01/MEAS_MON.U
- ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/CT36_10_01/MEAS_MON.U

These two nodes have their values changed every second on the OPC Server, but we are only receiving `datachange_notification` for the second node with the value of the first node.
Example:

> Tag_1 = { 
>     address: ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/TT36_10_01/MEAS_MON.U,
>     value: 30,8 
> }
> Tag_2 = { 
>    address: ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/CT36_10_01/MEAS_MON.U,
>     value: 0,11
> }
> 
> Output at subscription: `Tag_2.value = 30,8`

**Comment by AndreasHeine** [comment]
can you provide more logs?

what are the subscription parameter e.g. publishInterval?

what params do you pass to subscribe datachange? e.g. samplinginterval, filters, ...

 small code snipits would help too!

have you tested it with UaExpert does it work with it?

**Reply by JuliaCordeiro** [reply]
We are creating the subscription inside the `subscribe` function of OPCUA_SUB_HANDLER client.
This is example with some pieces of our code:

> self.polling_interval = 128
> self.client = Client(opc_ua_host_address)
> 
> subscription = await self.client.create_subscription( period=self.polling_interval, handler=self )
> await subscription.subscribe_data_change(self.nodes)

And about the OPC, we're using OPCExpert with OPC_UA protocol

**Reply by AndreasHeine** [reply]
have you tried a simplified test like in our examples?

e.g.:
https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples%2Fclient-subscription.py
https://github.com/AndreasHeine/opcua-tutorial/blob/main/client%2Fsubscribe.py

**Reply by JuliaCordeiro** [reply]
Yes, we're just logging inside the `datachange_notification` and doing a reset of our watchdog.

> def datachange_notification(self, node: Node, val, data):
>         self.logger.log_info(  f"DataChange Node: {node.__str__()}, val tag_reading: {val}", 200)
>         self.watchdog.reset()

**Reply by JuliaCordeiro** [reply]
We've noticed that in a very strange way, the `test_tag` that we insert at the first position of our `nodes_array` was causing the problem. But we couldn't find the reason for this behavior.
Example of our node_array with problem:
> nodes = [
> "ns=0;i=2258",
> "ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/TT36_10_01/MEAS_MON.U",
> "ns=3;s=167.252.15.79->OPCServer.WinCC.1>CPY_R0_PRJ_OSS1SB::Eq_Com_CFL_L1/AIs_1_CFL_L1/CT36_10_01/MEAS_MON.U"
> ]

Change the position of the `test_tag` for the middle of the array the error persist. 
We find out that using the `test_tag` at the last position the problem, apparently, is resolved. We would like to know why.

**Reply by AndreasHeine** [reply]
the thing is we just ask the server to add nodes (monitored items) to a subscription with a few parameters... we as client have no clue what the win cc opc ua server does with that!
